### PR TITLE
fix: implement percentile computation logic for SingleStore

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes:
- remove the dispatch logic in the profiler interface to allow override of metric computation methods
- override at runtime percentile metrics for SingleStore. We can't add a `compiles` logic as the dialect will be flagged as `mysql`. We go around this by implementing a SingleStore profiler interface and overwrite the window metrics method and ingesting SingleStore metrics at runtime

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [x] I have added a test that covers the exact scenario we are fixing. For complex issues, comment the issue number in the test for future reference.

### CODE REVIEW

File: ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py [hunk 1]
```diff
@@ -305,6 +286,9 @@ def _(
         Returns:
             dictionnary of results
         """
+        session = kwargs.get("session")
+        column = kwargs.get("column")
```
Comment (pmbrull) [inline ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:290]: if we need both the session and column can we add them as explicit parameters?

Comment (TeddyCr) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:290]: @pmbrull we can. I was trying to comply with the parent method signature -- as mypy was complaining. Let me know what you think

Comment (pmbrull) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:290]: then we could add them to the parent as well right? I imagine that any metric computation would always need the `session` and `columns` so we could define it as such in the abs

Comment (TeddyCr) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py:290]: Not necessarily. data lake does not need a session object and `column` is table-specific.

**Review by pmbrull — DISMISSED** [review]

**Pull request merged**